forms-flow-api/src/formsflow_api/models/filter.py

```diff
@@ -0,0 +1,111 @@
+"""This manages Filter information."""
+from __future__ import annotations
+
+from typing import List
+
+from formsflow_api_utils.utils.enums import FilterStatus
+from sqlalchemy import JSON, and_, or_
+from sqlalchemy.dialects.postgresql import ARRAY
+
+from formsflow_api.models.base_model import BaseModel
+from formsflow_api.models.db import db
+
+from .audit_mixin import AuditDateTimeMixin, AuditUserMixin
+
+
+class Filter(AuditDateTimeMixin, AuditUserMixin, BaseModel, db.Model):
+    """This class manages filter information."""
+
+    id = db.Column(db.Integer, primary_key=True)
+    tenant = db.Column(db.String, index=True, nullable=True)
+    name = db.Column(db.String, nullable=False)
+    description = db.Column(db.String, nullable=True)
+    resource_id = db.Column(db.String, nullable=True)
+    criteria = db.Column(JSON, nullable=True)
+    variables = db.Column(ARRAY(JSON), nullable=True)
+    properties = db.Column(JSON, nullable=True)
+    roles = db.Column(ARRAY(db.String), nullable=True, comment="Applicable roles")
+    users = db.Column(ARRAY(db.String), nullable=True, comment="Applicable users")
+    status = db.Column(db.String(10), nullable=True)
+
+    @classmethod
+    def find_all_active_filters(cls, tenant: str = None) -> List[Filter]:
+        """Find all active filters."""
+        query = cls.query.filter(Filter.status == str(FilterStatus.ACTIVE.value))
+        if tenant:
+            query = cls.query.filter(Filter.tenant == tenant)
```

Comment:
This should be `query = query.filter(Filter.tenant == tenant)` right ? for multitenant also we are expecting to return active filters for the corresponding tenant right? The current code will overwrite the initial query if `tenant` is true, returning all the tenant-specific filters despite the status.

Reply:
good catch. I missed that. Will update

Reply:
done